src/main/java/com/ericsson/ei/handlers/IdRulesHandler.java

```diff
@@ -49,15 +51,16 @@ public void setJmesPathInterface(JmesPathInterface jmesPathInterface) {
         this.jmesPathInterface = jmesPathInterface;
     }
 
-    public void runIdRules(RulesObject rulesObject, String event) {
+    public void runIdRules(RulesObject rulesObject, String event) throws MongoDBConnectionException {
         if (rulesObject != null && event != null) {
             JsonNode idsJsonObj = getIds(rulesObject, event);
             if (idsJsonObj != null && idsJsonObj.isArray()) {
                 for (final JsonNode idJsonObj : idsJsonObj) {
                     final String id = idJsonObj.textValue();
                     final List<String> aggregatedObjects = matchIdRulesHandler.fetchObjectsById(rulesObject, id);
-                    aggregatedObjects.forEach(
-                        aggregatedObject -> extractionHandler.runExtraction(rulesObject, id, event, aggregatedObject));
+                    for(String aggregatedObject : aggregatedObjects) {
```

Comment:
need this change?

Reply:
yes its required otherwise its asking compile time error by adding the try-catch in our case its required to throw the Exception
So i changed to For-each to For loop